docs/cloud.md

```diff
@@ -4,4 +4,81 @@ layout: default
+
+## Cloud Providers
+
+Explore the three prominent cloud providers that are renowned for their extensive public cloud offerings as well as robust support for private and hybrid cloud environments, making them crucial considerations in your cloud journey.
+
+* **Amazon Web Services (AWS)**: [AWS](https://aws.amazon.com/){:target="_blank"} is a comprehensive and widely adopted cloud platform offering a broad range of services, including compute, storage, databases, machine learning, and more. It provides a highly scalable and secure infrastructure, with extensive global coverage and a rich ecosystem of tools and services.
+* **Microsoft Azure**: [Microsoft Azure](https://azure.microsoft.com/){:target="_blank"} is a cloud computing platform that provides a wide array of services for building, deploying, and managing applications. It offers integration with Microsoft tools and technologies, robust support for hybrid scenarios, and a strong focus on enterprise capabilities.
+* **Google Cloud Platform (GCP)**: [GCP](https://cloud.google.com/gcp){:target="_blank"} provides a suite of cloud computing services, including compute, storage, networking, machine learning, and data analytics. It emphasizes data analytics, machine learning, and AI capabilities, with seamless integration with other Google services.
+
+## Testing
+
+Testing in the cloud shares similarities with traditional application testing but with additional benefits.
+
+* In case of **functional testing**, testers can provision and configure test environments quickly, replicate production-like environments, and leverage cloud-based testing tools. It enables efficient and scalable test execution, parallel test runs, and the ability to test from multiple geographic locations. Overall, functional testing in the cloud enhances agility and accelerates the testing process.
+* **Non-functional testing** is more important in the cloud due to the dynamic and scalable nature of cloud environments. It focuses on aspects such as performance, scalability, security, and resilience. Non-functional testing in the cloud allows for load testing at scale, assessing system performance under various scenarios. It also facilitates security testing, including vulnerability assessments and penetration testing, to ensure data protection and compliance. The cloud provides a robust foundation for non-functional testing, enabling comprehensive validation of application behavior and quality.
+
+{: .more }
+To learn more about Cloud Computing, one good source is <https://www.guru99.com/cloud-computing-for-beginners.html>{:target="_blank"}.
+
+{: .questions }
+> 1. What are the three main types of clouds in terms of ownership and security?
+> 1. Define public, private, and hybrid clouds and provide a brief explanation of their characteristics.
+> 1. What is the difference between IaaS, PaaS, and SaaS? Provide a concise definition for each.
```

Comment:
```suggestion
> 1. What are the differences between IaaS, PaaS, and SaaS? Provide a concise definition for each type.
```

Reply:
Updated, except for the last word. These are service models so that the type could be mixed with cloud types.